cfstore/cfstore_client.go

```diff
@@ -0,0 +1,132 @@
+	Name                string      `json:"name"`
+	Guid                string      `json:"guid"`
+	CreatedAt           string      `json:"created_at"`
+	UpdatedAt           string      `json:"updated_at"`
+	Free                bool        `json:"free"`
+	Description         string      `json:"description"`
+	ServiceGuid         string      `json:"service_guid"`
+	Extra               interface{} `json:"extra"`
+	UniqueId            string      `json:"unique_id"`
+	Public              bool        `json:"public"`
+	Active              bool        `json:"active"`
+	Bindable            bool        `json:"bindable"`
+	ServiceUrl          string      `json:"service_url"`
+	ServiceInstancesUrl string      `json:"service_instances_url"`
+}
+
+type Service struct {
+	Guid              string   `json:"guid"`
+	CreatedAt         string   `json:"created_at"`
+	UpdatedAt         string   `json:"updated_at"`
+	Label             string   `json:"label"`
+	Description       string   `json:"description"`
+	Active            bool     `json:"active"`
+	Bindable          bool     `json:"bindable"`
+	ServiceBrokerGuid string   `json:"service_broker_guid"`
+	PlanUpdateable    bool     `json:"plan_updateable"`
+	Tags              []string `json:"tags"`
+}
+
+type CFDataClient interface {
+	ListServicePlans() ([]ServicePlan, error)
+	ListServices() ([]Service, error)
+}
+
+var _ CFDataClient = &Client{}
+
+type Client struct {
+	Client *cfclient.Client
+}
+
+func (c *Client) ListServicePlans() ([]ServicePlan, error) {
+	var servicePlans []ServicePlan
+	requestUrl := "/v2/service_plans"
+	for {
+		var servicePlansResp cfclient.ServicePlansResponse
+		r := c.Client.NewRequest("GET", requestUrl)
+		resp, err := c.Client.DoRequest(r)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error requesting service plans")
+		}
+		resBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error reading service plans request:")
+		}
+		err = json.Unmarshal(resBody, &servicePlansResp)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error unmarshaling service plans")
+		}
+		for _, servicePlanResource := range servicePlansResp.Resources {
+			servicePlan := ServicePlan{}
+			servicePlan.Guid = servicePlanResource.Meta.Guid
+			servicePlan.CreatedAt = servicePlanResource.Meta.CreatedAt
+			servicePlan.UpdatedAt = servicePlanResource.Meta.UpdatedAt
+			servicePlan.Free = servicePlanResource.Entity.Free
+			servicePlan.Name = servicePlanResource.Entity.Name
+			servicePlan.Description = servicePlanResource.Entity.Description
+			servicePlan.ServiceGuid = servicePlanResource.Entity.ServiceGuid
+			servicePlan.Extra = servicePlanResource.Entity.Extra
+			servicePlan.UniqueId = servicePlanResource.Entity.UniqueId
+			servicePlan.Public = servicePlanResource.Entity.Public
+			servicePlan.Active = servicePlanResource.Entity.Active
+			servicePlan.Bindable = servicePlanResource.Entity.Bindable
+			servicePlans = append(servicePlans, servicePlan)
```

Comment:
Not a blocker, but you can also do:

```go
servicePlans = append(servicePlans, ServicePlan{
  Guid: servicePlanResource.Meta.Guid,
  CreatedAt: servicePlanResource.Meta.CreatedAt,
  UpdatedAt: servicePlanResource.Meta.UpdatedAt,
  ...
})
```

personally find it cleaner.

Reply:
Instead of

```go
servicePlan := ServicePlan{}
servicePlan.Guid = servicePlanResource.Meta.Guid
servicePlan.CreatedAt = servicePlanResource.Meta.CreatedAt
servicePlan.UpdatedAt = servicePlanResource.Meta.UpdatedAt
servicePlans = append(servicePlans, servicePlan)
```

Wasn't clear enough. Sorry!